activesupport/lib/active_support/core_ext/numeric/conversions.rb

```diff
@@ -134,6 +134,12 @@ def to_formatted_s(*args)
   deprecate to_formatted_s: :to_s
 end
 
-[Fixnum, Bignum, Float, BigDecimal].each do |klass|
-  klass.prepend(ActiveSupport::NumericWithFormat)
+# Ruby 2.4+ unifies Fixnum & Bignum into Integer.
+if Integer == Fixnum
```

Comment:
Thanks for the suggestions. That prepends Fixnum's ancestor, not Fixnum, so no go.

To avoid referencing Fixnum entirely, the simple route is checking `1.class == Integer`. I didn't litter the code with this (there are half a dozen other cases) because it's speculating how—and whether—Fixnum references will spam warnings. Poor trade for unknown benefit. There's little drawback to waiting to see what drops on Ruby trunk.